src/test/suite/editors/collectionDocumentsProvider.test.ts

```diff
@@ -20,6 +22,10 @@ const mockDocumentsAsJsonString = `[
 ]`;
 
 suite('Collection Documents Provider Test Suite', () => {
+  afterEach(function () {
```

Comment:
We need to think about which tests ui we want to use. Mocha differentiates keywords for bdd and tdd approaches: https://mochajs.org/#bdd

According to our settings we use tdd:
https://github.com/mongodb-js/vscode/blob/VSCODE-43/Show-status-bar-item-while-running-actions/src/test/suite/index.ts#L17-L21

And in tests we use `suite()` what matches the concept. But if we want to use `afterEach()` or `before()` we start mixing everything. I just was writing tests for `mongosh` integration and stumbled upon the same problem :) I wanted to have `before(), after() and context()`. The `context()` is an even more interesting case. Tests runner complains that `context is not a function`, what is weird since it should be a callback.

Reply:
hmmm agreed... let's give it a think... we can definitely do some things to make the tests a bit more consistent and pattern driven.